### PR TITLE
Refactor ProjectData and add runtime bool for complex/real runs

### PR DESCRIPTION
**Pull request by williamfgc**
## Proposed changes

- Refactor ProjectData class to meet coding standards
- Add `is_complex` static variable for start migration from compile-time `QMC_COMPLEX`
- Add unit test

Related to #4099 

## What type(s) of changes does this code introduce?

- Code style update (formatting, renaming)
- Refactoring (no functional changes, no api changes)
- Testing changes (e.g. new unit/integration/performance tests)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
Ubuntu 20, must pass CI

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by prckent** [comment]
_Somewhere_ (Slack, GitHub issue?) there is a better suggestion for the input flag. I recall complex_algorithms, which is less ambiguous.

File: src/Utilities/ProjectData.h [hunk 1]
```diff
@@ -23,118 +23,133 @@
...
   // The driver version of the project
   DriverVersion driver_version_;
+
+  /**
+   * @brief runtime variable to track real or complex runs using a single compiled qmcpack executable
+   * @todo remove/rethink its "const" nature
+   * true: complex run
+   * false: real run (default)
+   */
+  static const bool is_complex_;
```
Comment (PDoakORNL) [inline src/Utilities/ProjectData.h:150]: This shouldn't be static.  Especially in testing I should be able to construct both a real and a complex ProjectData depending on the input.

**Review by PDoakORNL — CHANGES_REQUESTED** [review]

File: src/Utilities/ProjectData.h (hunk 1 above)
Comment (ye-luo) [reply src/Utilities/ProjectData.h:150]: Agreed.

**Review by PDoakORNL — APPROVED** [review]

**Review by ye-luo — CHANGES_REQUESTED** [review]
I have difficulty to follow what is changed in this PR mostly clouded by renaming and function relocation.
Please make a separate PR with renaming first.

**Comment by williamfgc** [comment]
@ye-luo I marked this PR as Draft and won't be merged. I will open new PRs, thanks.

**Comment by prckent** [comment]
Summary of Slack discussion: complex_wavefunction for the input variable, isComplex() internally.

**Comment by ye-luo** [comment]
I think all the desired change from this PR has been handled by #4190 and #4198

**Pull request closed without merging**